pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

```diff
@@ -0,0 +1,1827 @@
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_ClearSkipList(t *testing.T) {
+	type fields struct {
+		verifyFunction             func(context context.Context, ref name.Reference, co *cosign.CheckOpts) ([]oci.Signature, bool, error)
+		fetchImageManifestFunction func(ref name.Reference, options ...remote.Option) (*remote.Descriptor, error)
+		skippedImages              map[string]bool
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   map[string]bool
+	}{
+		{
+			name: "clear single image in map",
+			fields: fields{
+
+				verifyFunction:             nil,
+				fetchImageManifestFunction: nil,
+				skippedImages: map[string]bool{
+					"sha256:sampleimagehash": true,
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "clear multiple images map",
+			fields: fields{
+				verifyFunction:             nil,
+				fetchImageManifestFunction: nil,
+				skippedImages: map[string]bool{
+					"sha256:sampleimagehash":  true,
+					"sha256:sampleimagehash1": true,
+				},
+			},
+			want: nil,
+		},
+		{
+			name: "clear on empty map",
+			fields: fields{
+				verifyFunction:             nil,
+				fetchImageManifestFunction: nil,
+				skippedImages:              map[string]bool{},
+			},
+			want: nil,
+		},
+		{
+			name: "clear on nil map",
+			fields: fields{
+				verifyFunction:             nil,
+				fetchImageManifestFunction: nil,
+				skippedImages:              nil,
+			},
+			want: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				verifyFunction:             tt.fields.verifyFunction,
+				fetchImageManifestFunction: tt.fields.fetchImageManifestFunction,
+				skippedImages:              tt.fields.skippedImages,
+			}
+			sigstore.ClearSkipList()
+			if !reflect.DeepEqual(sigstore.skippedImages, tt.want) {
+				t.Errorf("sigstore.skippedImages = %v, want %v", sigstore.skippedImages, tt.want)
+			}
+		})
+	}
+}
+
+func TestSigstoreimpl_ValidateImage(t *testing.T) {
+	type fields struct {
+		verifyFunction             func(context context.Context, ref name.Reference, co *cosign.CheckOpts) ([]oci.Signature, bool, error)
+		fetchImageManifestFunction func(ref name.Reference, options ...remote.Option) (*remote.Descriptor, error)
+		skippedImages              map[string]bool
+	}
+	type args struct {
+		ref name.Reference
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "validate image",
+			fields: fields{
+				verifyFunction: nil,
+				fetchImageManifestFunction: func(ref name.Reference, options ...remote.Option) (*remote.Descriptor, error) {
+					return &remote.Descriptor{
+						Manifest: []byte(`sometext`),
+					}, nil
+				},
+				skippedImages: nil,
+			},
+			args: args{
+				ref: func(d name.Digest, err error) name.Digest { return d }(name.NewDigest("example.com/sampleimage@sha256:5fb2054478353fd8d514056d1745b3a9eef066deadda4b90967af7ca65ce6505")),
```

Comment:
it is confusing to create a function on each arg, why not to have a struct that implements this interface and we set values there?